vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/VirtualMachineMO.java

```diff
@@ -794,6 +798,33 @@ public boolean createFullClone(String cloneName, ManagedObjectReference morFolde
         return false;
     }
 
+    private void setDiskProvsioningType(VirtualMachineRelocateSpec relocSpec, ManagedObjectReference morDs,
+                                        Storage.ProvisioningType diskProvisioningType) throws Exception {
+        if (diskProvisioningType == null){
+            return;
+        }
+        List<VirtualMachineRelocateSpecDiskLocator> relocateDisks = relocSpec.getDisk();
+        List<VirtualDisk> disks = this.getVirtualDisks();
+        for (VirtualDisk disk: disks){
```

Comment:
in any case, does the VM have multiple disks here ?

Reply:
I haven't seen a case,  but the getVirtualDisks() method returns a list, so just being safe here.